Scfmodule4/src/ArrOperation.java

```diff
@@ -0,0 +1,175 @@
+import java.util.ArrayList;
+
+/*
+ * This class is used to do some array oprations like
+ * find maximum length of mirror, fix X and Y, 
+ * count clumps and split array.
+ */
+public class ArrOperation {
+
+	/*
+	 * this method is used to return the size of the largest mirror section
+	 * found in the input array.
+	 * @param arrayOfNumber This is array of numbers.
+	 * @return This return the size of the largest mirror section.
+	 */
+	public static int lengthOfMirror(int[] arrayOfNumber) {
+		int length_mirror = 0;
+		int count = 0;
+
+		if (arrayOfNumber.length == 0) {
+			throw new AssertionError(" Array is empty, assertion error in method lengthOfMirror");
```

Comment:
Exception error message should be end-user friendly. Its not good to mention method names in exceptions.